cmd/gateway-main.go

```diff
@@ -293,6 +293,12 @@ func gatewayMain(ctx *cli.Context, backendType gatewayBackend) {
 		log.EnableQuiet()
 	}
 
+	// Fetch address option
+	gatewayAddr := ctx.GlobalString("address")
+	if gatewayAddr == ":"+globalMinioPort {
```

Comment:
This fix can be made better.. @vadmeste we need to use just like `mc`

```
        Before: setGlobalsFromContext,
```

This would set the value properly and then use globalMinioAddr instead..  this style is little fishy and its also code without comment on the purpose of handling code for Global and Local flag in this manner.